src/extra_rules.jl

```diff
@@ -179,8 +179,14 @@ end
 Base.view(t::Tangent{T}, inds) where T<:SVector = view(T(ChainRulesCore.backing(t.data)), inds)
 Base.getindex(t::Tangent{<:SVector, <:NamedTuple}, ind::Int) = ChainRulesCore.backing(t.data)[ind]
 
-function ChainRules.frule((_, ∂x), ::Type{SArray{S, T, N, L}}, x::NTuple{L,Any}) where {S, T, N, L}
-    SArray{S, T, N, L}(x), SArray{S}(∂x)
+function ChainRules.frule(
+    (_, ∂x)::Tuple{Any, Tangent{TUP}},
```

Comment:
```suggestion
    (_, ∂x)::Tuple{<:Any, Tangent{TUP}},
```
Do you not need the subtype thing here to avoid diagonal dispatch problems?

Reply:
You do not afaik, because tuple being covarient.
I have run into weird behavour with using `<:` inside tuple type parameters (but in theory it should just have no effect)